lib/control-connection.js

```diff
@@ -773,6 +812,74 @@ ControlConnection.prototype.reset = function (callback) {
   });
 };
 
+/**
+ * Computes the highest supported protocol version collectively by the given hosts.
+ * 
+ * Considers the cassandra_version of the input hosts to determine what protocol versions
+ * are supported and uses the highest common protocol version amont them.
+ * 
+ * If hosts >= C* 3.0 are detected, any hosts older than C* 2.1 will not be considered
+ * as those cannot be connected to.  In general this will not be a problem as C* does
+ * not support clusters with nodes that have versions that are more than one major
+ * version away from each other.
+ *
+ * @param {Number} version Maximum version to consider.
+ * @param {Array.<Host>} hosts The hosts to determine highest protocol version from.
+ * @return {Number} Highest supported protocol version among hosts.
+ */
+ControlConnection.prototype._highestCommonProtocolVersion = function(version, hosts) {
+  let maxVersion = version;
+  // whether or not protocol v3 is required (nodes detected that don't support < 3).
+  let v3Requirement = false;
```

Comment:
We _could_ use a `minVersion` field but its maybe an overkill, because dropping old protocol versions is not something that happens too often on the server side.

Reply:
Will think about how to make this better.  You are right that things will likely get messy next time a protocol version is dropped (C* 4.0?) so it would be good to take care of that now if there is an elegant way of doing that.

Reply:
I think for now I'd like to leave this as it is.  I think it would be better to address this next time protocol version support is removed from a C* version, and also we don't have to worry about this scenario as much because there shouldn't be nodes in the cluster separated by more than one version anyways.